types/react-native/index.d.ts

```diff
@@ -469,6 +471,84 @@ export interface Insets {
     right?: number;
 }
 
+export interface PressableStateCallbackType {
+    pressed: boolean;
+}
+
+export interface PressableProps extends AccessibilityProps, Omit<ViewProps, 'style' | 'hitSlop'> {
+    /**
+     * Called when a single tap gesture is detected.
+     */
+    onPress?: (event: GestureResponderEvent) => void;
+
+    /**
+     * Called when a touch is engaged before `onPress`.
+     */
+    onPressIn?: (event: GestureResponderEvent) => void;
+
+    /**
+     * Called when a touch is released before `onPress`.
+     */
+    onPressOut?: (event: GestureResponderEvent) => void;
+
+    /**
+     * Called when a long-tap gesture is detected.
+     */
+    onLongPress?: (event: GestureResponderEvent) => void;
+
+    /**
+     * Either children or a render prop that receives a boolean reflecting whether
+     * the component is currently pressed.
+     */
+    children: React.ReactNode | ((state: PressableStateCallbackType) => React.ReactNode);
+
+    /**
+     * Duration (in milliseconds) from `onPressIn` before `onLongPress` is called.
+     */
+    delayLongPress?: number;
+
+    /**
+     * Whether the press behavior is disabled.
+     */
+    disabled?: boolean;
+
+    /**
+     * Additional distance outside of this view in which a press is detected.
+     */
+    hitSlop?: Insets | number;
```

Comment:
```suggestion
    hitSlop?: null | Insets | number;
```

Reply:
Might also be a good idea to [follow upstream](https://github.com/facebook/react-native/blob/4f897336b6ce6c640a65f86b65c6d386ea55b944/Libraries/StyleSheet/Rect.js#L20) and define a `RectOrSize` type.